src/components/utils/src/threads/thread_posix.cc

```diff
@@ -89,15 +89,18 @@ void* Thread::threadFunc(void* arg) {
   thread->state_lock_.Acquire();
   thread->state_cond_.Broadcast();
 
-  while (!thread->finalized_) {
+  while (!thread->IsSwitchOnState(kThreadStateFinalize)) {
     LOG4CXX_DEBUG(logger_, "Thread #" << pthread_self() << " iteration");
     thread->run_cond_.Wait(thread->state_lock_);
-    LOG4CXX_DEBUG(logger_,
-                  "Thread #" << pthread_self() << " execute. "
-                             << "stopped_ = " << thread->stopped_
-                             << "; finalized_ = " << thread->finalized_);
-    if (!thread->stopped_ && !thread->finalized_) {
-      thread->isThreadRunning_ = true;
+    LOG4CXX_DEBUG(
+        logger_,
+        "Thread #" << pthread_self() << " execute. "
+                   << "stopped_ = " << thread->IsSwitchOnState(kThreadStateStop)
+                   << "; finalized_ = "
+                   << thread->IsSwitchOnState(kThreadStateFinalize));
+    if (!thread->IsSwitchOnState(kThreadStateStop) &&
+        !thread->IsSwitchOnState(kThreadStateFinalize)) {
```

Comment:
This check is present above(about `FINALIZE` state). Is it necessary here?

Reply:
@YarikMamykin 
Yes, because flag kThreadStateStop maybe reset in start function.